### PR TITLE
Ft/basic server // payment 

### PR DESCRIPTION
**Pull request by incogni23**
worked on issue 1 and 2

### CODE REVIEW

**Review by shubham-aurva — APPROVED** [review]

**Comment by incogni23** [comment]
worked on config APIs

File: auth/service.go [hunk 1]
```diff
@@ -32,32 +36,35 @@ func NewDBVar(d Dao) AuthServicer {
 	}
 }
 
-func (dbv *DBVar) SignUp(incomingUser User) error {
+func (dbv *DBVar) SignUp(incomingUser *User) (*User, error) {
```
Comment (shubham-aurva) [inline auth/service.go]: name it just `service`

File: auth/service.go [hunk 2]
```diff
@@ -32,32 +36,35 @@ func NewDBVar(d Dao) AuthServicer {
...
+		return nil, errors.New("username, password, and email are required")
+	}
+
 	existingUser, err := dbv.db.GetUser(incomingUser.Username)
 	if existingUser != nil && existingUser.Username != "" {
-		return errors.New("User already exists")
+		return nil, errors.New("User already exists")
 	}
 
 	if err != nil {
```
Comment (shubham-aurva) [inline auth/service.go:56]: put this condition on top

File: services/monolith/configurations/config.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	DB *gorm.DB
+}
+
+func NewDBConfig(db *gorm.DB) *DBConfig {
+	return &DBConfig{
+		DB: db,
+	}
+}
+
+func (dbc *DBConfig) PaymentMethod(c *gin.Context) {
```
Comment (shubham-aurva) [inline services/monolith/configurations/config.go]: GetPaymentMethod should be the name

File: services/monolith/configurations/config.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
+package configurations
```
Comment (shubham-aurva) [inline services/monolith/configurations/config.go]: file name should be endpoint.go

File: services/monolith/main.go [hunk 5]
```diff
@@ -16,8 +20,17 @@ func main() {
...
-
-	db.AutoMigrate(&auth.User{})
+	
+	db.AutoMigrate(&auth.User{},
+		&payments.PaymentMethod{},
+		&payments.PaymentGateway{},
+		&order.Order{},
+		&payments.Payment{},
+		&payments.ThirdPartyToken{},
+		&payments.PaymentConfiguration{},
```
Comment (shubham-aurva) [inline services/monolith/main.go]: this shouldn't be the part of global db

File: services/monolith/dependencies/seed.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+	if res.RowsAffected == 0 {
+		log.Printf("Payment configuration already exists")
+	} else {
+		log.Printf("Payment configuration successful")
+	}
+
+	return nil
+}
+
+func SeedPayment(db *gorm.DB) {
```
Comment (shubham-aurva) [inline services/monolith/dependencies/seed.go]: name it either Seed or something on the lines of `SeedPaymentProperties`

File: services/monolith/dependencies/seed.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package dependencies
+
+import (
+	"log"
+
+	"github.com/monolith/payments"
+	"gorm.io/gorm"
+)
+
+func Initconfig(db *gorm.DB, config payments.PaymentConfiguration) error {
```
Comment (shubham-aurva) [inline services/monolith/dependencies/seed.go]: remove this

File: services/monolith/main.go [hunk 8]
```diff
@@ -31,6 +44,12 @@ func main() {
 
 	unprotectedGroup.AuthGroup.POST("/login", authEndpoint.Login)
 
+	configService := configurations.NewDBConfig(db)
+
+	r.GET("/payment_method/:method_id", configService.PaymentMethod)
```
Comment (shubham-aurva) [inline services/monolith/main.go]: this should be called via protectedGroup

File: services/monolith/configurations/config.go [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+	var paymentConfig payments.PaymentConfiguration
+
+	err := c.ShouldBindJSON(&paymentConfig)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	err = dbc.DB.Create(&paymentConfig).Error
```
Comment (shubham-aurva) [inline services/monolith/configurations/config.go]: So there will be few checks before inserting this data to database
The first check is to validate that strings are not empty (PAYMENT METHOD AND PAYMENT GATEWAT != "")

The second check is to validate that PAYMENT GATEWAY is actually exisiting in our database (ensuring we have a support it) and we should be able to query the id as well

The third check is to validate that PAYMENT METHOD is actually exisiting in our database (ensuring we have a support it)

Comment (shubham-aurva) [reply services/monolith/configurations/config.go]: if any of them fails, we should result an error

**Review by shubham-aurva — CHANGES_REQUESTED** [review]

**Review by mishtuu — APPROVED** [review]

**Pull request merged**